### PR TITLE
Create id set - add the option to create an ID set on a specific pack

### PR DESCRIPTION
**Pull request by ChanochShayner**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/31175

## Description
add the option to create an ID set on a specific pack.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by guykeller — CHANGES_REQUESTED** [review]
please add tests

File: demisto_sdk/commands/create_id_set/tests/update_id_set_test.py [hunk 1]
```diff
@@ -47,6 +47,12 @@ def test_create_id_set_output(self):
         id_set_creator.create_id_set()
         assert os.path.exists(self.file_path)
 
+    def test_create_id_set_on_specific_pack(self):
+        id_set_creator = IDSetCreator(self.file_path, input='Packs/AMP')
+
+        id_set_creator.create_id_set()
+        assert os.path.exists(self.file_path)
```
Comment (yaakovi) [inline demisto_sdk/commands/create_id_set/tests/update_id_set_test.py:54]: add more checks here:
* the pack elements were collected
* there aren't any other element (from other packs)

**Review by yaakovi — COMMENTED** [review]
sync with @abaumgarten as she changed the id set too in https://github.com/demisto/demisto-sdk/pull/911

File: demisto_sdk/commands/create_id_set/tests/update_id_set_test.py (hunk 1 above)
Comment (yaakovi) [reply demisto_sdk/commands/create_id_set/tests/update_id_set_test.py:54]: @ChanochShayner 
what about this case
> there aren't any other element (from other packs)

**Comment by ChanochShayner** [comment]
@yaakovi Does what I added cover the case you mentioned?

File: demisto_sdk/commands/create_id_set/tests/update_id_set_test.py [hunk 2]
```diff
@@ -70,6 +76,42 @@ def test_create_id_set_no_output(self, mocker):
         assert 'Widgets' in id_set.keys()
         assert 'Mappers' in id_set.keys()
 
+    def test_create_id_set_on_specific_pack(self, repo):
```
Comment (guykeller) [inline demisto_sdk/commands/create_id_set/tests/update_id_set_test.py:79]: 1. Let's add given - when - then to all the tests, you can check for examples in content repo
2. Let's try to work on tests readability. Make sure it is clear what the flow is and what is being tested + the inner logic of each test

**Review by guykeller — CHANGES_REQUESTED** [review]

**Review by yaakovi — COMMENTED** [review]
> @yaakovi Does what I added cover the case you mentioned?

yes, looks ok.
merge after approval from @guykeller

File: demisto_sdk/commands/create_id_set/tests/update_id_set_test.py [hunk 3]
```diff
@@ -70,6 +88,67 @@ def test_create_id_set_no_output(self, mocker):
...
+        """
+        pack = repo.create_pack()
+
+        id_set_creator = IDSetCreator(self.file_path, pack.path)
+
+        id_set_creator.create_id_set()
+
+        with open(self.file_path, 'r') as id_set_file:
+            private_id_set = json.load(id_set_file)
+        for _, entity_value in private_id_set.items():
```
Comment (guykeller) [inline demisto_sdk/commands/create_id_set/tests/update_id_set_test.py:149]: `entity_value` - unclear name

File: demisto_sdk/commands/create_id_set/tests/update_id_set_test.py [hunk 4]
```diff
@@ -70,6 +88,67 @@ def test_create_id_set_no_output(self, mocker):
         assert 'Widgets' in id_set.keys()
         assert 'Mappers' in id_set.keys()
 
+    def test_create_id_set_on_specific_pack(self, repo):
+        """
+        Given
+        - two packs with integrations to create an ID set from them
```
Comment (guykeller) [inline demisto_sdk/commands/create_id_set/tests/update_id_set_test.py:94]:
```suggestion
        - two packs with integrations to create an ID set from
```

File: demisto_sdk/commands/create_id_set/tests/update_id_set_test.py [hunk 5]
```diff
@@ -70,6 +88,67 @@ def test_create_id_set_no_output(self, mocker):
...
+
+        When
+        - create ID set on pack1
```
Comment (guykeller) [inline demisto_sdk/commands/create_id_set/tests/update_id_set_test.py:97]: unclear line

File: demisto_sdk/commands/create_id_set/tests/update_id_set_test.py [hunk 6]
```diff
@@ -70,6 +88,67 @@ def test_create_id_set_no_output(self, mocker):
...
+
+        Then
+        - ensure there is only one integration in the ID set integrations list - integration1
+        - ensure output id_set contains pack_to_create_id_set_on - integration1
+        - ensure output id_set does not contain the pack_to_not_create_id_set_on - integration2
```
Comment (guykeller) [inline demisto_sdk/commands/create_id_set/tests/update_id_set_test.py:102]: These lines should not reflect the names of variables in the code.
They should provide a high level explanation of what the test tests, and why

File: demisto_sdk/commands/create_id_set/tests/update_id_set_test.py [hunk 7]
```diff
@@ -70,6 +88,67 @@ def test_create_id_set_no_output(self, mocker):
...
+
+        """
+        packs = repo.packs
+
+        pack_to_create_id_set_on = repo.create_pack('pack1')
```
Comment (guykeller) [inline demisto_sdk/commands/create_id_set/tests/update_id_set_test.py:107]: `pack1`, `integration1`, etc - aren't informative names

**Review by guykeller — APPROVED** [review]
Fix the comments I've added, generalise them for the other tests and done

File: demisto_sdk/commands/create_id_set/tests/update_id_set_test.py (hunk 3 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/create_id_set/tests/update_id_set_test.py]: done

File: demisto_sdk/commands/create_id_set/tests/update_id_set_test.py (hunk 4 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/create_id_set/tests/update_id_set_test.py]: done

File: demisto_sdk/commands/create_id_set/tests/update_id_set_test.py (hunk 5 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/create_id_set/tests/update_id_set_test.py]: done

File: demisto_sdk/commands/create_id_set/tests/update_id_set_test.py (hunk 6 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/create_id_set/tests/update_id_set_test.py]: done

File: demisto_sdk/commands/create_id_set/tests/update_id_set_test.py (hunk 7 above)
Comment (ChanochShayner) [reply demisto_sdk/commands/create_id_set/tests/update_id_set_test.py]: done

**Pull request merged**